apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -165,7 +166,14 @@ export default class LibraryPublisher extends React.Component {
       libraryJson,
       error => {
         console.warn(`Error publishing library: ${error}`);
-        this.setState({publishState: PublishState.ERROR_PUBLISH});
+        if (
+          error.message ===
```

Comment:
what does this error object look like? could we match on a status code or something instead of matching the message string?

Reply:
Yeah, I thought about that too but it's not clear to me how to do this more cleanly. This is was we have at this point:
![image](https://user-images.githubusercontent.com/8787187/108140182-69e50280-7076-11eb-87bb-d856a6615e31.png)
It comes from [here](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/initApp/clientApi.js#L233)

let me know if you know a better way to do this?

Reply:
oh, right, i've run into this same thing before. i don't think there's a better way to handle this right now 😢  i added a tech debt ticket for us to discuss: https://codedotorg.atlassian.net/browse/STAR-1441